misc/protobuf-ballerina/src/main/java/org/ballerinalang/protobuf/cmd/GrpcCmd.java

```diff
@@ -99,52 +102,69 @@
     
     @Override
     public void execute() {
-        
-        File protoFile = new File(protoPath);
-        if (!protoFile.isFile() || !protoFile.exists() || EMPTY_STRING.equals(protoPath) ||
-                !protoPath.contains(".proto")) {
-            String errorMessage = "Invalid proto file location. Please input valid proto file location.";
+
+        if (protoPath == null || !protoPath.toLowerCase(Locale.ENGLISH).endsWith(PROTO_SUFFIX)) {
+            String errorMessage = "Invalid proto file path. Please input valid proto file location.";
             outStream.println(errorMessage);
             throw new BalGenToolException(errorMessage);
         }
-        protoPath = protoFile.getAbsolutePath();
+
+        if (!Files.isReadable(Paths.get(protoPath))) {
+            String errorMessage = "Provided service proto file is not readable. Please input valid proto file " +
+                    "location.";
+            outStream.println(errorMessage);
+            throw new BalGenToolException(errorMessage);
+        }
+
         if (helpFlag) {
-            String commandUsageInfo = BLauncherCmd.getCommandUsageInfo(parentCmdParser, "build");
+            String commandUsageInfo = BLauncherCmd.getCommandUsageInfo("build");
```

Comment:
Shall we use a constant for "build"

Reply:
Fixed in f2bb0753488e0c357b61ea5a9d3639e43bdf9bfe